### PR TITLE
fix: NPE in OptionObjectBundleHook.preUpdate

### PR DESCRIPTION
**Pull request by vietnguyen**
https://dhis2.atlassian.net/browse/DHIS2-13640
- According to the bug reported, user got NPE when trying to add `Option` to an `OptionSet`.
- I couldn't reproduce it but the issue can be fixed in backend with a `null` check.

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/option/OptionSet.java [hunk 1]
```diff
@@ -122,6 +123,16 @@ public void removeAllOptions()
         options.clear();
     }
 
+    public void removeOption( Option option )
+    {
+        if ( options == null || CollectionUtils.isEmpty( options ) )
```
Comment (jbee) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/option/OptionSet.java:128]: Main reason to use a static `isEmpty` is that it allows for `null` so it does not need a `null` check before

Comment (vietnguyen) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/option/OptionSet.java]: fixed

**Review by mortenoh — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/option/OptionSet.java [hunk 2]
```diff
@@ -122,6 +123,14 @@ public void removeAllOptions()
         options.clear();
     }
 
+    public void removeOption( Option option )
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/option/OptionSet.java:126]: This is a little bit worrying as we might be hiding underlying issues. Do we know why the options list is null here?

**Review by larshelge — APPROVED** [review]

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/option/OptionSet.java (hunk 2 above)
Comment (vietnguyen) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/option/OptionSet.java:126]: @larshelge  I talked with Pamod yesterday. They can't reproduce the issue after restarting the sever. All OptionSets are now working normally on their instance. 
My theory is that after executing the clean_up_option_sortOrder script, it reset all the OptionSet.options references in the cache. But I couldn't replicate this behaviour on my local. So not really sure what happened on their instance.
I will update the clean up script document to add a final step for restarting server.

**Pull request merged**